core/domain/job_services_test.py

```diff
@@ -0,0 +1,332 @@
+
+        with self.set_job_request_post_callback(post_callback):
+            with self.assertRaisesRegexp(
+                Exception,
+                'Expected answer_group_index to be a key in training_data'):
+                job_services.get_next_job()
+
+        job_data['training_data'] = [{'answer_group_index': 1}]
+        post_callback = self._get_post_callback_for_next_job(job_data)
+
+        with self.set_job_request_post_callback(post_callback):
+            with self.assertRaisesRegexp(
+                Exception,
+                'Expected answers to be a key in training_data'):
+                job_services.get_next_job()
+
+        job_data['training_data'] = [{
+            'answer_group_index': '1',
+            'answers': ['a', 'b', 'c']
+        }]
+        post_callback = self._get_post_callback_for_next_job(job_data)
+
+        with self.set_job_request_post_callback(post_callback):
+            with self.assertRaisesRegexp(
+                Exception,
+                'Expected answer_group_index to be an int'):
+                job_services.get_next_job()
+
+        job_data['training_data'] = [{
+            'answer_group_index': 1,
+            'answers': 'answer1'
+        }]
+        post_callback = self._get_post_callback_for_next_job(job_data)
+        with self.set_job_request_post_callback(post_callback):
+            with self.assertRaisesRegexp(
+                Exception,
+                'Expected answers to be a list'):
+                job_services.get_next_job()
+
+    def test_job_validation_raises_exception_on_invalid_algorithm_version(self):
+        """Test that job validation raises exception if algorithm_version is
+        invalid in the received job data.
+        """
+        job_data = {
+            'job_id': '1',
+            'algorithm_id': 'TextClassifier',
+            'training_data': [],
+            'algorithm_version': '123'
+        }
+        post_callback = self._get_post_callback_for_next_job(job_data)
+
+        with self.set_job_request_post_callback(post_callback):
+            with self.assertRaisesRegexp(
+                Exception, 'Expected algorithm version to be integer'):
+                job_services.get_next_job()
+
+    def test_job_validation_raises_exception_on_unknown_algorithm_id(self):
+        """Test that job validation raises exception if algorithm_id is
+        unknown in the received job data.
+        """
+        job_data = {
+            'job_id': '1',
+            'algorithm_id': 'ab',
+            'training_data': [],
+            'algorithm_version': 1
+        }
+        post_callback = self._get_post_callback_for_next_job(job_data)
+
+        with self.set_job_request_post_callback(post_callback):
+            with self.assertRaisesRegexp(
+                Exception, 'Invalid algorithm id ab'):
+                job_services.get_next_job()
+
+    def test_job_validation_raises_exception_on_unknown_algorithm_version(self):
```

Comment:
Maybe instead: 'incorrect' version would be clearer?

Reply:
Done